### PR TITLE
Document serialization in rustc

### PR DESCRIPTION
**Pull request by matthewjasper**
This documents serialization after rust-lang/rust#73851

### CODE REVIEW

File: src/serialization.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# Serialization in Rustc
+
+Rustc has to [serialize] and deserialize various data during compilation.
+Specifially:
```
Comment (MajorBreakfast) [inline src/serialization.md:4]:
```suggestion
Specifically:
```

**Comment by matthewjasper** [comment]
@mark-i-m This is now ready to be merged

File: src/serialization.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+For types that are constructed from those types, `Encodable` and `Decodable` are
+usually implemented by [derives]. These generate implementations that forward
+deserialization to the fields of the struct or enum. For a struct those impls
+look something like this:
+
+```rust
+# #![feature(rustc_private)]
+# extern crate rustc_serialize;
+# use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
```
[the remark this answers is not in the document]
Comment (JohnTitor) [reply src/serialization.md:41]: Hm, this isn't addressed yet?

[the remark this answers is not in the document]
Comment (mark-i-m) [reply src/serialization.md:41]: I think it should work on the latest mdbook... this was an old bug.

[the remark this answers is not in the document]
Comment (mark-i-m) [reply src/serialization.md:41]: i can just do it in a followup

**Pull request merged**